rasa/model.py

```diff
@@ -275,64 +310,19 @@ def persist_fingerprint(output_path: Text, fingerprint: Fingerprint):
     dump_obj_as_json_to_file(path, fingerprint)
 
 
-def core_fingerprint_changed(
-    fingerprint1: Fingerprint, fingerprint2: Fingerprint
+def section_fingerprint_changed(
+    fingerprint1: Fingerprint, fingerprint2: Fingerprint, section: Section
 ) -> bool:
-    """Checks whether the fingerprints of the Core model changed.
-
-    Args:
-        fingerprint1: A fingerprint.
-        fingerprint2: Another fingerprint.
-
-    Returns:
-        `True` if the fingerprint for the Core model changed, else `False`.
-
-    """
-    relevant_keys = [
-        FINGERPRINT_CONFIG_KEY,
-        FINGERPRINT_CONFIG_CORE_KEY,
-        FINGERPRINT_DOMAIN_KEY,
-        FINGERPRINT_STORIES_KEY,
-        FINGERPRINT_RASA_VERSION_KEY,
-    ]
-
-    for k in relevant_keys:
-        if fingerprint1.get(k) != fingerprint2.get(k):
-            logger.info("Data ({}) for Core model changed.".format(k))
-            return True
-    return False
-
-
-def nlu_fingerprint_changed(
-    fingerprint1: Fingerprint, fingerprint2: Fingerprint
-) -> bool:
-    """Checks whether the fingerprints of the NLU model changed.
-
-    Args:
-        fingerprint1: A fingerprint.
-        fingerprint2: Another fingerprint.
-
-    Returns:
-        `True` if the fingerprint for the NLU model changed, else `False`.
-
-    """
-
-    relevant_keys = [
-        FINGERPRINT_CONFIG_KEY,
-        FINGERPRINT_CONFIG_NLU_KEY,
-        FINGERPRINT_NLU_DATA_KEY,
-        FINGERPRINT_RASA_VERSION_KEY,
-    ]
-
-    for k in relevant_keys:
+    """Check whether the fingerprint of a section has changed."""
+    for k in section.relevant_keys:
         if fingerprint1.get(k) != fingerprint2.get(k):
-            logger.info("Data ({}) for NLU model changed.".format(k))
+            logger.info("Data ({}) for {} section changed.".format(k, section.name))
```

Comment:
mhm, I can't decide between `debug` and `info`. What do you think?
```suggestion
            logger.info("Data ({}) for {} section changed.".format(k, section.name))
```

Reply:
I always prefer to err on the side of verbosity, so I'd still vouch for `info`.